src/prommis/uky/uky_flowsheet.py

```diff
@@ -861,5 +873,956 @@ def display_results(m):
+        costing_method=QGESSCostingData.get_REE_costing,
+        costing_method_arguments={
+            "cost_accounts": R_roaster_accounts,
+            "scaled_param": m.fs.R_roaster.duty,
+            "source": 1,
+            "n_equip": 1,
+            "scale_down_parallel_equip": False,
+            "CE_index_year": CE_index_year,
+        },
+    )
+
+    # 3.4 is UKy Roasting - Gas Scrubber
+    R_gas_scrubber_accounts = ["3.4"]
+    m.fs.R_gas_scrubber = UnitModelBlock()
+    m.fs.R_gas_scrubber.gas_rate = Var(
+        initialize=11.500, units=units.ft**3 / units.min
+    )
+    m.fs.R_gas_scrubber.gas_rate.fix()
+    m.fs.R_gas_scrubber.costing = UnitModelCostingBlock(
+        flowsheet_costing_block=m.fs.costing,
+        costing_method=QGESSCostingData.get_REE_costing,
+        costing_method_arguments={
+            "cost_accounts": R_gas_scrubber_accounts,
+            "scaled_param": m.fs.R_gas_scrubber.gas_rate,
+            "source": 1,
+            "n_equip": 1,
+            "scale_down_parallel_equip": False,
+            "CE_index_year": CE_index_year,
+        },
+    )
+
+    # 3.5 is UKy Roasting - Spray Chamber Quencher (7000-60000 ft**3/min)
+    R_spray_chamber_quencher_accounts = ["3.5"]
+    m.fs.R_spray_chamber_quencher = UnitModelBlock()
+    m.fs.R_spray_chamber_quencher.gas_rate = Var(
+        initialize=11.500, units=units.ft**3 / units.min
+    )
+    m.fs.R_spray_chamber_quencher.gas_rate.fix()
+    m.fs.R_spray_chamber_quencher.costing = UnitModelCostingBlock(
+        flowsheet_costing_block=m.fs.costing,
+        costing_method=QGESSCostingData.get_REE_costing,
+        costing_method_arguments={
+            "cost_accounts": R_spray_chamber_quencher_accounts,
+            "scaled_param": m.fs.R_spray_chamber_quencher.gas_rate,
+            "source": 1,
+            "n_equip": 3,
+            "scale_down_parallel_equip": False,
+            "CE_index_year": CE_index_year,
+        },
+    )
+
+    # 3.7 is UKy Roasting - Chiller
+    R_chiller_accounts = ["3.7"]
+    m.fs.R_chiller = UnitModelBlock()
+    m.fs.R_chiller.duty = Var(initialize=13.1, units=units.MBTU / units.hr)
+    m.fs.R_chiller.duty.fix()
+    m.fs.R_chiller.costing = UnitModelCostingBlock(
+        flowsheet_costing_block=m.fs.costing,
+        costing_method=QGESSCostingData.get_REE_costing,
+        costing_method_arguments={
+            "cost_accounts": R_chiller_accounts,
+            "scaled_param": m.fs.R_chiller.duty,
+            "source": 1,
+            "n_equip": 1,
+            "scale_down_parallel_equip": False,
+            "CE_index_year": CE_index_year,
+        },
+    )
+
+    feed_input = units.convert(
+        flowsheet.fs.leach_solid_feed.flow_mass[0]
+        + flowsheet.fs.leach_liquid_feed.flow_vol[0]
+        * flowsheet.fs.leach.liquid[0, 1].dens_mol,
+        to_units=units.ton / units.hr,
+    )
+    # Feed grade is grams of solute per 1000 grams of solution (solution ~ water since dilute concentrations)
```

Comment:
The feed grade is the concentration of REE in the feedstock ore. This flowsheet has no Crushing & Screening or Pre-Roasting operations, so it would be based on the solid flow into the leaching unit. Basically, we're assuming Crushing & Screening and Pre-Roasting have no losses since we don't have those units here.

Reply:
So just to be clear, feed grade would only be the concentration of REEs, meaning it would exclude the concentrations of the contaminants (Al, Fe, and Ca)? And would it be the concentration of the REE or the REE oxide?

Reply:
Yes, feed grade would only be the concentration of REEs in the original bulk solid, where contaminants, solid waste, ash, etc are part of the bulk solid but only saleable products are considered as REEs. And yes, REE not REE oxide.

Reply:
Maybe I'm just misunderstanding, but isn't ppm a weird/incorrect unit of measurement to be using for a solid feed stream (i.e. we can't get the REE concentration in mg/mL for the solid feed, but we could get something like kg/hr). So I'm a bit confused what the units of feed grade should be here.

Reply:
No - remember that ppm is an application specific unit. Here it means micrograms per gram and not mg/L.